docs/api/navigators/StackNavigator.md

```diff
@@ -118,7 +118,7 @@ All `navigationOptions` for the `StackNavigator`:
 
 - `title` - a title (string) of the scene
 - `header` - a config object for the header bar:
-  - `visible` - Boolean toggle of header visibility. Only works when `headerMode` is `screen`.
+  - `bar` - React Element to display as a header or `null` to hide it. Only works when `headerMode` is `screen`
```

Comment:
@satya164 problem with `headerComponent` is that it's more on a low-level side of things. One should be able to override header for a single scene without implementing entire header once again (there's a lot of stuff, like renderXComponent that has to be implemented).

I believe that most people will not even read from `router` or `navigation` as their use-cases will be rather simple. However I agree we can leave it to avoid limiting too much.